### PR TITLE
feat(bankholidays): service to retrieve bank holidays for country, year

### PR DESCRIPTION
**Pull request by Steffy29**
New service to retrieve bank holidays for country and year
It's possible to check if a date is a bank holiday for selected country
Ref: UXCT-19

### CODE REVIEW

File: src/bank-holidays/bank-holidays.constant.js [hunk 1]
```diff
@@ -0,0 +1,230 @@
+export const BANK_HOLIDAYS = {
```
Comment (FredericEspiau) [inline src/bank-holidays/bank-holidays.constant.js]: Don't we have an API for this ? Isn't there a library for this ?

Comment (FredericEspiau) [reply src/bank-holidays/bank-holidays.constant.js]: Wouldn't this work ? https://github.com/commenthol/date-holidays

Comment (Steffy29) [reply src/bank-holidays/bank-holidays.constant.js]: I created this service to to have the bank holidays which will be used by two controllers into the manager telecom (TelecomTelephonyAliasPortabilityOrderCtrl and TelephonySchedulerBankHolidaysCtrl).

Comment (FredericEspiau) [reply src/bank-holidays/bank-holidays.constant.js]: I understand what you did :)

Don't you think this shouldn't be done by us ? Does anyone in the API team work on a solution to be able to retrieve the bank holidays from them ?

**Comment by jleveugle** [comment]
Hello @Steffy29,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Comment by Steffy29** [comment]
I'm waiting action/decision from you. 
Don't close it without explanation about it.

File: src/bank-holidays/bank-holidays.service.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    I -= (Math.floor(I / 28) * (1 - (Math.floor(I / 28) * Math.floor(29 / (I + 1)) * Math.floor((21 - N) / 11))));
+    let J = Y + Math.floor(Y / 4) + I + 2 - C + Math.floor(C / 4);
+    J -= (7 * Math.floor(J / 7));
+    const L = I - J;
+    const M = 3 + Math.floor((L + 40) / 44);
+    const D = L + 28 - (31 * Math.floor(M / 4));
+
+    return [Y, _.padLeft(M, 2, '0'), _.padLeft(D, 2, '0')].join('-');
+  }
+  /* eslint-enable */
```
[the remark this answers is not in the document]
Comment (Steffy29) [reply src/bank-holidays/bank-holidays.service.js]: OK but I need this feature for getting bank holidays displayed on scheduler to disable the ability to choose days which are bank holidays for portability (before the upgrade it generated failure for portability). 
So what do we do with this feature and the bug which is linked with it?

**Comment by antleblanc** [comment]
@Steffy29 I noticed that you have to extract this part of the code in order to
be able to use it from another controller that helps you to fix an existing bug.

But as @FredericEspiau and @marie-j says, we have to improve this extracted
part and find a proper and maintainable solution.

To move forward on that subject I will merge this pull request to fix the
existing bug. But feel free to open new pull request which will improve this feature.

Thanks.

**Pull request merged**